lib/membrane_rtc_engine/events/request_track_variant.ex

```diff
@@ -21,9 +21,11 @@ defmodule Membrane.RTC.Engine.Event.RequestTrackVariant do
   Type describing RequestTrackVariant event.
 
   * `variant` - track variant to receive
+  * `reason` - reason of the request. It will be repeated in
+  `Membrane.RTC.Engine.Event.TrackVariantSwitched`. Defaults to `nil`.
```

Comment:
I am not sure about including the `reason` in events but:
* it was the easiest way of storing the reason of request
* it doesn't break compatibility as by default is `nil`
* it doesn't introduce any complex logic in Tee
* it's harder to manage variant request reasn in TrackReceiver as there might be the following case:
```
Tee <- request :medium - TR
Tee -> give :medium <- request :low in the same time - TR (here is RC)
Tee -> give :low - TR  
```
as a result TR can get two TrackVariantSwitched events in a row so saving the last `reason` isn't probably enough?

Reply:
I think that makes sense to pass on the reason why we are requesting another track variant.